scripts/train/train.py

```diff
@@ -1,149 +1,71 @@
-def main(arguments):
-    parser = argparse.ArgumentParser(
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    # environment parameters
-    # parser.add_argument(
-    #     '--data-folder',
-    #     help="local path to training data",
-    #     required=True
-    # )
-
-    parser.add_argument(
-        "--output-dir", type=str,
-        default=os.path.join('..', '..', 'data', 'training', 'outputs'),
-        help='location to write output'
-    )
-
-    # training specific parameters
-    parser.add_argument('--kernel', type=str, default='linear',
-                        help='Kernel type to be used in the algorithm')
-    parser.add_argument('--penalty', type=float, default=1.0,
-                        help='Penalty parameter of the error term')
-
-    # parse the arguments
-    args = parser.parse_args(arguments)
-
-    # setup output directory
-    # model_output_dir = os.path.join(
-    #     os.path.dirname(os.path.realpath(__file__)),
-    #     args.output_dir)
-    # os.makedirs(args.output-dir, exist_ok=True)
-
-    train(args.output_dir, args.kernel, args.penalty)
-
-
-if __name__ == '__main__':
-    sys.exit(main(sys.argv[1:]))
+import random
+import warnings
+from pathlib import Path
+import spacy
+from spacy.util import minibatch, compounding
+
+
+
+def main(model=None,TRAIN_DATA=None , output_dir=None, n_iter=100):
+    # Use built-in "ner" pipeline components
+    nlp = spacy.load("ro_core_news_lg")
+    ner = nlp.get_pipe("ner")
+
+    # add labels
+    for _, annotations in TRAIN_DATA:
+        for ent in annotations.get("entities"):
+            ner.add_label(ent[2])
+
+    print("Started Training")
+    # get names of other pipes to disable them during training
+    pipe_exceptions = ["ner", "trf_wordpiecer", "trf_tok2vec"]
+    other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in pipe_exceptions]
+    # only train NER
+    with nlp.disable_pipes(*other_pipes), warnings.catch_warnings():
+        # show warnings for misaligned entity spans once
+        warnings.filterwarnings("once", category=UserWarning, module='spacy')
+
+        # reset and initialize the weights randomly – but only if we're
+        # training a new model
+        if model is None:
+            nlp.begin_training()
+        for itn in range(n_iter):
+            random.shuffle(TRAIN_DATA)
+            losses = {}
+            # batch up the examples using spaCy's minibatch
+            batches = minibatch(TRAIN_DATA, size=compounding(4.0, 32.0, 1.001))
+            for batch in batches:
+                texts, annotations = zip(*batch)
+                nlp.update(
+                    texts,  # batch of texts
+                    annotations,  # batch of annotations
+                    drop=0.5,  # dropout - make it harder to memorise data
+                    losses=losses,
+                )
+            print("Losses", losses)
+
+    # test the trained model
+    for text, _ in TRAIN_DATA:
+        doc = nlp(text)
```

Comment:
pt ce mai faci acest for loop daca doc nu se foloseste mai departe?

Reply:
removed